packages/next/src/build/templates/middleware.ts

```diff
@@ -27,10 +27,12 @@ function errorHandledHandler(fn: AdapterOptions['handler']) {
       return await fn(...args)
     } catch (err) {
       const req = args[0]
+      const url = new URL(req.url)
+      const resource = url.href.slice(url.origin.length)
```

Comment:
rather than slicing off the origin, I think making it more explicit that this is `url.pathname + url.search` is a bit clearer.